cmake/preproject.cmake

```diff
@@ -11,8 +11,13 @@ if((NOT CMAKE_C_COMPILER)
    AND "$ENV{CC}" STREQUAL ""
    AND "$ENV{CXX}" STREQUAL ""
 )
-  find_program(FOUND_CMAKE_C_COMPILER NAMES clang-15 clang-11)
-  find_program(FOUND_CMAKE_CXX_COMPILER NAMES clang++-15 clang++-11)
+  if("${COMPILE_TARGET}" STREQUAL "sgx")
+    find_program(FOUND_CMAKE_C_COMPILER NAMES clang-11)
+    find_program(FOUND_CMAKE_CXX_COMPILER NAMES clang++-11)
+  else()
+    find_program(FOUND_CMAKE_C_COMPILER NAMES clang-15)
+    find_program(FOUND_CMAKE_CXX_COMPILER NAMES clang++-15)
+  endif()
```

Comment:
Separate change that feels necessary. Otherwise `CC` and `CXX` have to be specified to `cmake` when building CCF from source.